Scripts/Haptics/XRNodeHapticPulse.cs

```diff
@@ -0,0 +1,85 @@
+﻿namespace VRTK.Core.Haptics
+{
+    using UnityEngine;
+    using UnityEngine.XR;
+
+    /// <summary>
+    /// Creates a timed haptic pulse on a <see cref="XRNode"/> supported haptic device.
+    /// </summary>
+    public class XRNodeHapticPulse : HapticPulse
+    {
+        /// <summary>
+        /// The device to pulse.
+        /// </summary>
+        [Tooltip("The device to pulse.")]
+        public XRNode device = XRNode.LeftHand;
+        /// <summary>
+        /// The duration to pulse the device for.
+        /// </summary>
+        [Tooltip("The duration to pulse the device for.")]
+        public float duration = 0.005f;
+
+        /// <summary>
+        /// The haptic capabilities of the specified <see cref="XRNode"/>.
+        /// </summary>
+        protected HapticCapabilities caps = new HapticCapabilities();
+
+        /// <inheritdoc />
+        public override bool IsActive()
+        {
+            return base.IsActive();
+        }
+
+        /// <inheritdoc />
+        protected override void DoBegin()
+        {
+            Pulse(Intensity, duration, device);
+        }
+
+        /// <inheritdoc />
+        protected override void DoCancel()
+        {
+            Pulse(0, 0, device);
+        }
+
+        /// <summary>
+        /// Sends a pulse to the current device.
+        /// </summary>
+        /// <param name="intensity">The intensity to pulse with.</param>
+        /// <param name="duration">The duration to pulse for.</param>
+        /// <param name="node">The device to pulse on.</param>
+        protected virtual void Pulse(float intensity, float duration, XRNode node)
+        {
+            if (!InputDevices.GetDeviceAtXRNode(node).TryGetHapticCapabilities(out caps))
```

Comment:
Put the output of `InputDevices.GetDeviceAtXRNode(node)` into a local variable before this `if` and reuse it in the whole method, that way we only call this method once.